internal/checks/checks.go

```diff
@@ -307,6 +310,8 @@ func (c *Updater) loop(ctx context.Context) error {
 	c.logger.Info().Int64("probe id", c.probe.Id).Str("probe name", c.probe.Name).Msg("registered probe with synthetic-monitoring-api")
 
 	c.metrics.connectionStatus.Set(1)
+	// true indicates that probe was able to connect to API
+	c.IsReady.Store(true)
```

Comment:
while I agree that this does what we need, I would prefer to see a function here:

```
c.IsConnected(true)
defer c.IsConnected(false)
```

so what needs to be stored in the Updater struct is:

```
    IsConnected func(bool)
```

same for the options.

At the location where the options are passed, you can have:

```
IsConnected: func(v bool) { isReady.Set(v) }
```

this allows you to propagate the `connected` state without embedding too much information as to why you need to know that state elsewhere. For example, elsewhere you could decide that "is ready" means "has been connected for 1 second" or "it connected to the API once" (or whatever other criteria is appropriate).

Reply:
updated